src/interface_r/R/kmeans.R

```diff
@@ -0,0 +1,39 @@
+#' K-Means clustering
+#' 
+#' @export
+h2o4gpu.kmeans <- function(
+  n_clusters = 8L,
+  init = 'k-means++',
+  n_init = 1L,
+  max_iter = 300L,
+  tol = 1e-4,
+  precompute_distances = 'auto',
+  verbose = 0,
+  random_state = NULL,
+  copy_x = TRUE,
+  n_jobs = 1,
+  algorithm = 'auto',
+  gpu_id = 0,
+  n_gpus = -1,
+  do_checks = 1,
+  backend = 'auto'
+) {
+  model <- h2o4gpu$KMeans(
```

Comment:
So the user will pass in data via `%>% fit(x)`? Is this the behavior we want? I would think we want to pass in `X` directly with the rest of the params much like other R functions.

Reply:
We can but it's nicer to separate them conceptually, e.g. initialize the model object and then decide what to do with it (fit, predict, transform, etc.). One use case is to load a pre-trained model and transform new data or continue training using new data (you only initialize the model once and then use it for multiple purposes, e.g. transform and fit). It's more consistent with sklearn, keras, etc.

Reply:
Note that you can also do `fit(model, x)`.

Reply:
Sounds good to me!